addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,264 @@
+        :return: None
+        """
+        if not products:
+            return
+
+        ICP = self.env['ir.config_parameter']
+        session = requests.Session()
+        custom_search_req = requests.Request(
+            method='GET',
+            url='https://customsearch.googleapis.com/customsearch/v1',
+            params={
+                'cx': ICP.get_param('google.pse.id').strip(),
+                'safe': 'active',
+                'searchType': 'image',
+                'key': ICP.get_param('google.custom_search.key').strip(),
+                'rights': 'cc_publicdomain,cc_attribute,cc_sharealike',
+                'imgSize': 'large',
+                'imgType': 'photo',
+                'fields': 'searchInformation/totalResults,items(link)',
+            },
+        )
+        prepared_req = session.prepare_request(custom_search_req)
+        request_url = prepared_req.url
+        for product in products:
+            try:
+                prepared_req.prepare_url(request_url, {'q': product.barcode})
+
+                response = session.send(prepared_req, timeout=5)
+                response.raise_for_status()
+                response_content = response.json()
+            except (ConnectTimeout, ReadTimeout):
+                # If the timeout is reached for one product, we still need to process the others
+                continue
+            except HTTPError as http_error:
+                # Sometimes Google returns a service_unavailable error which is temporary.
+                # In that case, the product shall not be marked as processed and should be fetched
+                # again later.
+                if http_error.response.status_code == requests.codes.service_unavailable:
+                    continue
+                else:
+                    raise
+            else:
+                if int(response_content['searchInformation']['totalResults']) > 0:
+                    for item in response_content['items']:
+                        if 'x-raw-image:///' in item['link']:  # Ignores images with an incorrect link
+                            continue
+                        try:
+                            req = session.get(item['link'], timeout=5)
+                            req.raise_for_status()
+                            if 'image/' in req.headers['Content-Type']:
+                                product.image_1920 = base64.b64encode(req.content)
+                                break
+                        except Exception:
+                            # No matter what error happens here, it means that the image is either
+                            # unreachable or not base64 encodable. As we need an image for this
+                            # product, we continue to process the next link in the list.
+                            continue
+            finally:
+                product.image_fetch_pending = False
+                self.env.cr.commit()  # Commit every image in case the cron is killed
```

Comment:
Implicitly cancelling (and committing) the fetch on a timeout or an error seems odd.

Reply:
We need to make sure that image_fetch_pending is false if we didn't found an appropriate image to avoid indefinitely fetch the same image. Sometimes, an error could mean that the format isn't supported by odoo so the product has no valid image.